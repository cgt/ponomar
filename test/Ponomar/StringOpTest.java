package Ponomar;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.io.ByteArrayOutputStream;
import java.io.PrintStream;

public class StringOpTest {
    private final PrintStream stdout = System.out;
    private final ByteArrayOutputStream capturedOutput = new ByteArrayOutputStream();

    @BeforeEach
    public void setUp() {
        System.setOut(new PrintStream(capturedOutput));
    }

    @AfterEach
    public void tearDown() {
        System.setOut(stdout);
    }

    @Test
    public void main() {
        StringOp.main(new String[]{""});
        Approvals.verify(capturedOutput.toString());
    }
}