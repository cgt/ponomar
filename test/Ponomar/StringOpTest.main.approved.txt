In testing.dayInfo we have { nday=3; wday=6; ndayP=81;  }
In testin2.dayInfo, we have { nday=3; wday=6; ndayP=81; testing=-123;  }
